src/services/gitlab/GitLabService.ts

```diff
@@ -0,0 +1,468 @@
+   */
+  async listPullRequests(
+    owner: string,
+    repo: string,
+    options?: { withDiffStat?: boolean } & ListGetterOptions<{ state?: PullRequestState }>,
+  ): Promise<Paginated<PullRequest>> {
+    const state = VCSServicesUtils.getGitLabPRState(options?.filter?.state);
+
+    const { data, pagination } = await this.unwrap(
+      this.client.MergeRequests.list(`${owner}/${repo}`, {
+        pagination: options?.pagination,
+        filter: { state },
+      }),
+    );
+
+    const user = await this.getUserInfo(owner);
+
+    const items: PullRequest[] = await Promise.all(
+      data.map(async (val) => {
+        const pullRequest = {
+          user: {
+            id: val.author.id.toString(),
+            login: val.author.username,
+            url: val.author.web_url,
+          },
+          url: val.web_url,
+          body: val.description,
+          createdAt: val.created_at.toString(),
+          updatedAt: val.updated_at.toString(),
+          closedAt: val.closed_at ? val.closed_at?.toString() : null,
+          mergedAt: val.merged_at ? val.merged_at?.toString() : null,
+          state: val.state,
+          id: val.iid, //id?
+          base: {
+            repo: {
+              url: `${this.host}/${owner}/${repo}`,
+              name: repo,
+              id: val.project_id.toString(),
+              owner: user,
+            },
+          },
+        };
+        // Get number of changes, additions and deletions in PullRequest if the withDiffStat is true
+        if (options?.withDiffStat) {
+          //TODO
+          // https://gitlab.com/gitlab-org/gitlab/issues/206904
+          const lines = await this.getPullsDiffStat(owner, repo, val.iid);
+          return { ...pullRequest, lines };
+        }
+        return pullRequest;
+      }),
+    );
+    const customPagination = this.getPagination(pagination);
+
+    return { items, ...customPagination };
+  }
+
+  /**
+   * Get a single pull request.
+   */
+  async getPullRequest(owner: string, repo: string, prNumber: number, withDiffStat?: boolean): Promise<PullRequest> {
+    const { data } = await this.unwrap(this.client.MergeRequests.get(`${owner}/${repo}`, prNumber));
+    const user = await this.getUserInfo(owner);
+
+    const pullRequest: PullRequest = {
+      user: {
+        id: data.author.id.toString(),
+        login: data.author.username,
+        url: data.author.web_url,
+      },
+      url: data.web_url,
+      body: data.description,
+      createdAt: data.created_at.toString(),
+      updatedAt: data.updated_at.toString(),
+      closedAt: data.closed_at ? data.closed_at?.toString() : null,
+      mergedAt: data.merged_at ? data.merged_at?.toString() : null,
+      state: data.state,
+      id: data.iid,
+      base: {
+        repo: {
+          url: `${this.host}/${owner}/${repo}`,
+          name: repo,
+          id: data.project_id.toString(),
+          owner: user,
+        },
+      },
+    };
+
+    //TODO
```

Comment:
TODO what? Please describe what's necessary to do.

Reply:
Sorry, I just forgot about it there.